### PR TITLE
Payment forms in the deployment flow

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- if you know what you're doing, open the deployment flow and if you have no user for this stuff you should see the form

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6997/

File: jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js [hunk 1]
```diff
@@ -89,8 +93,252 @@ YUI.add('deployment-payment', function() {
...
+            disabled={disabled}
+            label="Address line 1"
+            required={true} />
+          <juju.components.GenericInput
+            disabled={disabled}
+            label="Address line 2 (optional)"
+            required={false} />
+          <juju.components.GenericInput
+            disabled={disabled}
+            label="County (optional)"
```
Comment (fabricematrat) [inline jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js]: I think calling it State/Province might be less confusing, I had so many confusion in the payment service.

File: jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js [hunk 2]
```diff
@@ -89,8 +93,252 @@ YUI.add('deployment-payment', function() {
...
+        </div>);
+    },
+
+    /**
+      Generate the business name field if required.
+
+      @method _generateBusinessNameField
+    */
+    _generateBusinessNameField: function() {
+      if (!this.state.business) {
```
Comment (fabricematrat) [inline jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js]: May be we need to make sure if a business name is entered, a VAT is provided ?

File: jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js [hunk 3]
```diff
@@ -89,8 +93,252 @@ YUI.add('deployment-payment', function() {
...
+                disabled={disabled}
+                label="Postcode"
+                required={true} />
+            </div>
+            <div className="four-col">
+              <juju.components.InsetSelect
+                disabled={disabled}
+                label="Country code"
+                onChange={null}
+                options={[]} />
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js]: I'm guessing this gets fetched in the other branch?

File: jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js (hunk 2 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js]: That would only be necessary for countries in which VAT is a thing :)

File: jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js (hunk 3 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js]: Yep!

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7016/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**